### ISSUE TITLE
Erreur 500 

### ISSUE THREAD

**Issue by tdhk69** [bokub/conso-api#16]
Bonjour,
A partir de votre application j'arrive à activer le consentement pour le partage de mes données et je récupère le token sans problème. 
Mais lorsque je veux lancer un test (sur n'importe quelle type de donnée ou période) j'obtiens une erreur 500.

Je remarque aussi que dans mon compte Enedis je n'ai aucun tiers associé au partage de mes données malgré l'activation du consentement.
Est-ce que le problème provient d'Enedis ?
Je précise aussi que mon PRM est uniquement configuré en mode production. 

```
{
  "status": 500,
  "message": "The Enedis API returned an error",
  "error": {
    "error": "technical_error",
    "error_description": "Technical error. Please try later."
  }
}
```

**Comment by PaddyRu** [comment]
Bonjour,
Je suis dans le même cas de figure, avez-vous réussi depuis ?

**Comment by tdhk69** [comment]
Bonjour 
Non toujours pas.
Je viens de réessayer l'activation du consentement et j'ai maintenant une erreur "Compteur incompatible", même plus possibilité de récupérer le token. 
J'ai essayé aussi avec d'autres services (MyElectricalData) et ça ne marche pas non plus, donc le problème semble bien venir d'Enedis.

**Comment by bokub** [comment]
Hello,

> Est-ce que le problème provient d'Enedis ?

Oui, complètement

> Je remarque aussi que dans mon compte Enedis je n'ai aucun tiers associé au partage de mes données malgré l'activation du consentement.

ça c'est normal

>  j'ai maintenant une erreur "Compteur incompatible"

Tu peux me faire une capture d'écran de toute la page s'il te plait ? (tu peux flouter tes infos persos) 


@PaddyRu Est-ce que toi aussi tu es uniquement en mode "production" ?

**Comment by PaddyRu** [comment]
Non, j’ai les 2 modes et les deux requêtes finissent en erreur 500.

**Comment by tdhk69** [comment]
> Tu peux me faire une capture d'écran de toute la page s'il te plait ? (tu peux flouter tes infos persos)
> 

Hé bien aujourd'hui je n'ai plus l'erreur de "Compteur incompatible", j'ai donc pu récupérer un nouveau token.
Mais toujours l'erreur 500 quand je veux l'utiliser... 

<img width="676" alt="image" src="https://github.com/bokub/conso-api/assets/132200691/fb4d96af-1c9d-4243-b230-48ce9c557d63">

**Comment by tdhk69** [comment]
Oui, même chose que toi @PaddyRu , j'avais aussi testé Hello Watt avec succès.

**Comment by bokub** [comment]
Hello,
C'est "normal" de ne rien voir dans l'interface de partage des données, Enedis n'applique pas le même traitement à mon API qu'à Ecowatt ou Total 🤷🏻‍♂️

**Comment by PaddyRu** [comment]
Et du coup on se débloque comment ?

**Comment by bokub** [comment]
Pour info, @gpion a eu le même souci dans #41 et son erreur a disparu.

Pour ceux qui ont le problème, pouvez-vous réessayer svp ?

**Comment by PaddyRu** [comment]
J'ai testé avec l'ancien token, toujours en erreur 500. Puis j'ai généré un nouveau token, mais également en erreur 500 :/

**Comment by tdhk69** [comment]
Pas mieux non plus. J'ai essayé aussi avec l'outil en ligne de commande au cas où, mais même erreur.

J'ai contacté par mail le service Data Connect (dataconnect@enedis.fr) pour expliquer mon problème. Je ne crois pas trop à une réponse mais peut-être que plusieurs demandes augmenteront les chances ;-)

**Comment by GaboFlo** [comment]
Pour info, j'ai aussi des 500 sur deux tokens qui ont été créés il y a déjà plusieurs semaines
Wait & see ...

`{
    "status": 500,
    "message": "The Enedis API returned an error",
    "error": "500 Internal Server Error"
}`

**Comment by bokub** [comment]
> Enedis vous donnerait il accès à un outil pour remonter les bugs de manière plus formelle ?

Non, tout ce que Enedis m'envoie vous l'avez en réponse de conso API, je n'ai **rien** de plus. C'est pour ca que je suis assez impuissant dans cette affaire

**Comment by YttraZZ** [comment]
Bonjour à tous,

J'ai un compteur prod et un compteur conso.
J'ai bien évidemment le meme probleme que vous, mais avec une différence.

J'avais déja reussi à synchroniser mon compteur conso avant Noel. Tout marche bien depuis.

J'ai collé aux basques d'Enedis car je ne pouvais pas ajouter mon compteur prod au site client Enedis.

Ca marche depuis quelques jours. J'ai pu générer un token pour les deux compteurs, et quand je teste j'ai une erreur 500.
 Mais .... quand j'utilise le token dans ha-linky, j'ai bien la synchro de mon compte conso. La synchro du compteur prod semble fonctionner aussi mais le log indique qu'il n'y a pas de données à charger.

Ce qui est bizarre c'est que les données sont bien visibles sur le site enedis.

Mes questions :

Suis-je au bon endroit ?
Ai-je raté qquechose dans ma conf ?
As tu besoin de qque chose ? (Log,conf,...)

Merci d'avance

**Comment by PaddyRu** [comment]
> > Enedis vous donnerait il accès à un outil pour remonter les bugs de manière plus formelle ?
> 
> Non, tout ce que Enedis m'envoie vous l'avez en réponse de conso API, je n'ai **rien** de plus. C'est pour ca que je suis assez impuissant dans cette affaire

Pouvez-vous me communiquer les API Enedis que vous appelez ? Les personnes avec lesquelles j'essaye de résoudre mon souci chez Enedis ne veulent pas comprendre que l'erreur 500 est renvoyée par eux ...

**Comment by YttraZZ** [comment]
> Bonjour à tous,
> 
> J'ai un compteur prod et un compteur conso. J'ai bien évidemment le meme probleme que vous, mais avec une différence.
> 
> J'avais déja reussi à synchroniser mon compteur conso avant Noel. Tout marche bien depuis.
> 
> J'ai collé aux basques d'Enedis car je ne pouvais pas ajouter mon compteur prod au site client Enedis.
> 
> Ca marche depuis quelques jours. J'ai pu générer un token pour les deux compteurs, et quand je teste j'ai une erreur 500. Mais .... quand j'utilise le token dans ha-linky, j'ai bien la synchro de mon compte conso. La synchro du compteur prod semble fonctionner aussi mais le log indique qu'il n'y a pas de données à charger.
> 
> Ce qui est bizarre c'est que les données sont bien visibles sur le site enedis.
> 
> Mes questions :
> 
> Suis-je au bon endroit ? Ai-je raté qquechose dans ma conf ? As tu besoin de qque chose ? (Log,conf,...)
> 
> Merci d'avance

J'ai résolu mon probleme en generant à nouveau mes deux tokens (conso et prod).

Cette fois-ci tout marche. Je ne touche plus rien

**Comment by tifriis208** [comment]
Bonjour,

Nouvel utilisateur via le module ha-linky, je rencontre aussi l'erreur 500 :

> Connection with Home Assistant established
> [16/12 09:20] Synchronization started for consumption data
> Last saved statistic date is 11/12/2024
> Cannot fetch consumption load curve from 2024-12-12 to 2024-12-16, here is the error:
> Conso API a répondu avec une erreur
> Code: 500
> Réponse : {
>     "status": 500,
>     "message": "The Enedis API returned an error",
>     "error": {
>         "error": "technical_error",
>         "error_description": "Technical error. Please try later."
>     }
> }
> Data import returned nothing !
> Connection with Home Assistant closed

Pourtant, via le site https://conso.boris.sh/exemples, j'arrive bien à récupérer les données liées à mon token.

J'ai tenté une remise à zéro (reset), mais ça persiste.

Auriez-vous une piste ? d'avance merci

**Comment by PaddyRu** [comment]
Pour ma part cela s'est résolu lorsque j'ai enfin eu des personnes compétentes au bout du fil.
J'avais donc déjà mon compteur ajouté dans l'espace client Enedis depuis quelques temps, mais suite à l'installation des panneaux photovoltaïque et dû à la présence de ma femme également sur le contrat EDF, il y a eu un mélangeage de pinceaux dans les données échangées entre EDF et Enedis qui ont mené à un bazar interne chez Enedis, résultant en erreurs 500 sur les nouveaux tokens (Hello-Watt présent avant tout ces changements a toujours fonctionné par exemple).
Il me fallait donc contacter EDF (mon fournisseur d'énergie, éventuellement un autre pour vous), afin qu'il envoie à Enedis les bonnes données pour que les données de mon profil Enedis matchent avec celles transmises par EDF à Enedis.
Il est certainement aussi possible de faire coïncider les données côté Enedis (modifiables depuis l'espace client directement si je ne dis pas de bêtises), afin de les faire correspondre avec ce qu'envoie le fournisseur d'énergie, pour moi ce n'était pas une option car en parallèle j'ai notre résidence secondaire sur le même compte EDF et Enedis et pour celui-ci les remontées EDF étaient correctes et matchaient avec mes données côté Enedis ...

**Comment by tdhk69** [comment]
> Pour ma part cela s'est résolu lorsque j'ai enfin eu des personnes compétentes au bout du fil. J'avais donc déjà mon compteur ajouté dans l'espace client Enedis depuis quelques temps, mais suite à l'installation des panneaux photovoltaïque et dû à la présence de ma femme également sur le contrat EDF, il y a eu un mélangeage de pinceaux dans les données échangées entre EDF et Enedis qui ont mené à un bazar interne chez Enedis, résultant en erreurs 500 sur les nouveaux tokens (Hello-Watt présent avant tout ces changements a toujours fonctionné par exemple). Il me fallait donc contacter EDF (mon fournisseur d'énergie, éventuellement un autre pour vous), afin qu'il envoie à Enedis les bonnes données pour que les données de mon profil Enedis matchent avec celles transmises par EDF à Enedis. Il est certainement aussi possible de faire coïncider les données côté Enedis (modifiables depuis l'espace client directement si je ne dis pas de bêtises), afin de les faire correspondre avec ce qu'envoie le fournisseur d'énergie, pour moi ce n'était pas une option car en parallèle j'ai notre résidence secondaire sur le même compte EDF et Enedis et pour celui-ci les remontées EDF étaient correctes et matchaient avec mes données côté Enedis ...

Merci pour l'info. Je suis dans le même cas : contrat EDF aux 2 noms et contrat Enedis seulement à mon nom...
